### PR TITLE
Implement level, fix #32

### LINKED ISSUE #32: **Bug:** If you let the game keep running in the BG, the rain algo goes wild and spawns too many objs.
- a simple mitigation could be to find some DOM event that fires when the game loses focus and use that to pause the game

### PR DESCRIPTION
**Pull request by tranhuuphucuong**
- Implement level
- FIx #32, Added `onWindowVisibilityChange` function.
- Fix some rendering function.

### CODE REVIEW

File: falling.js [hunk 1]
```diff
@@ -74,9 +75,29 @@ function fall(yPos, speed, rate, onFalling) {
...
+    7: 50,
+    8: 100,
+    9: 150,
+    10: 200,
+}
 function generateObject(logic) {
     // TODO: generate number based on logic.state.gameLevel
-    const n = [randomInRange(1, 5), randomInRange(1, 5)].sort()
+
+    logic.state.gameLevel
```
Comment (poltak) [inline falling.js:93]: This line doesn't need to be here.

Cool idea with the game levels btw!

File: falling.js [hunk 2]
```diff
@@ -103,25 +125,30 @@ export function rain(logic) {
...
+            speed = logic.state.gameLevel * 20 + 300
 
-            const fallingLogic = fall(obj.yPos, speed, 60, (newYPos) => {
+            const fallingLogic = fall(obj.yPos, speed, 30, (newYPos) => {
                 obj.yPos = newYPos
                 logic.mutate(mutatorFns.setFallingObjPosY, obj, newYPos)
-                if (isColliding(logic.state.basket, obj)) {
+                const isCollide = isColliding(logic.state.basket, obj)
+
+                if (isCollide || obj.yPos > window.innerHeight) {
```
Comment (poltak) [inline falling.js:135]: When does `obj.yPos > window.innerHeight` become true?

**Review by poltak — APPROVED** [review]
The level system work is really cool! The pause on window visiblity change also seems to work good. 

LGTM

File: falling.js [hunk 3]
```diff
@@ -74,9 +75,29 @@ function fall(yPos, speed, rate, onFalling) {
...
+    9: 150,
+    10: 200,
+}
 function generateObject(logic) {
     // TODO: generate number based on logic.state.gameLevel
-    const n = [randomInRange(1, 5), randomInRange(1, 5)].sort()
+
+    logic.state.gameLevel
+    let maxDenominator = levelToMaxDenominatorMapper[logic.state.gameLevel]
+    let maxNominator = maxDenominator / 2
```
Comment (timotiousPrime) [inline falling.js:95]: I think the variable should be called maxNumerator

**Review by timotiousPrime — APPROVED** [review]

**Pull request merged**